WebApi/WebApiTest/ControllerTest/PostControllerTest.cs

```diff
@@ -0,0 +1,279 @@
+            string in_title, string in_content, int in_likesCount, bool in_isLiked, bool in_isPromoted)
+        {
+            //Arrange
+            var mockService = new Mock<IPostService>();
+            mockService.Setup(x => x.GetById(0)).Returns(new PostDTO
+            {
+                id = in_id,
+                author = in_author,
+                authorID = in_authorID,
+                category = in_category,
+                title = in_title,
+                content = in_content,
+                likesCount = in_likesCount,
+                datetime = in_time,
+                isLikedByUser = in_isLiked,
+                isPromoted = in_isPromoted
+
+            });
+
+            var mockLogger = new Mock<ILogger<PostController>>();
+            var controller = new PostController(mockLogger.Object, mockService.Object);
+
+            var expected = new PostDTO
+            {
+                id = in_id,
+                author = in_author,
+                authorID = in_authorID,
+                category = in_category,
+                title = in_title,
+                content = in_content,
+                likesCount = in_likesCount,
+                datetime = in_time,
+                isLikedByUser = in_isLiked,
+                isPromoted = in_isPromoted
+            };
+
+            //Act
+            var actual = controller.Get(userID, 0);
+
+            //Assert
+            Assert.Equal(expected.id, actual.id);
+            Assert.Equal(expected.author, actual.author);
+            Assert.Equal(expected.authorID, actual.authorID);
+            Assert.Equal(expected.category, actual.category);
+            Assert.Equal(expected.title, actual.title);
+            Assert.Equal(expected.content, actual.content);
+            Assert.Equal(expected.likesCount, actual.likesCount);
+            Assert.Equal(expected.datetime, actual.datetime);
+            Assert.Equal(expected.isLikedByUser, actual.isLikedByUser);
+            Assert.Equal(expected.isPromoted, actual.isPromoted);
+        }
+
+
+        [Theory]
+        [InlineData("Title title1","Content", 1, false)]
+        [InlineData("Title title title","Conte nt1 ", 2, false)]
+        [InlineData("Title"," Co ntent2 ", 4, true)]
+        [InlineData("Title","Conten t3", 3, true)]
+
+        //TODO: I dont know how to test it yet.
+        public void AddPost_Test(string in_title, string in_content, int categoryID, bool in_isPromoted)
+        {
+            //Arrange
+            //var mockService = new Mock<IPostService>();
+            //mockService.Setup(x => x.AddPostAsync(It.IsAny<PostEditDTO>(), It.IsAny<int>())).Returns(Task.Run(() =>
+            //{
+            //    return
+            //}));
+            //var mockLogger = new Mock<ILogger<PostController>>();
+            //var controller = new PersonController(mockLogger.Object, mockService.Object);
+            Assert.True(true);
+        }
```

Comment:
Tak na prawdę nie jesteś w stanie sprawdzić niczego poza tym co funkcja zwraca. Samym działaniem dodawania zajmą się testy integracyjne. Ty tylko piszesz co chcesz żeby funkcja zwróciła i to sprawdzasz